static assertion error on libsanitizer (mips64-unknown, n64 ABI)

crosstool-NG version: commit 3ac6f1791af77b682b1de4ad56157653aa566417
host OS: Debian 11 (gcc 10.2.0)
config: [config.gz](https://github.com/crosstool-ng/crosstool-ng/files/8697518/config.gz)
build log: [build.log.gz](https://github.com/crosstool-ng/crosstool-ng/files/8697551/build.log.gz)

Hi,

I'm trying to build mips64-unknown-linux-gnu cross compiler (GCC 11.3.0) with
libgomp and libsanitizer enabled. No experimental features are enabled.

At final gcc stage, I got static assertion error on libsanitizer:

```
[ALL  ]    In file included from /home/bagas/cross/workdir/mips64-unknown/.build/mips64-unknown-linux-gnu/src/gcc/libsanitizer/sanitizer_common/sanitizer_platform_limits_linux.cpp:21:
[ERROR]    /home/bagas/cross/workdir/mips64-unknown/.build/mips64-unknown-linux-gnu/src/gcc/libsanitizer/sanitizer_common/sanitizer_platform_limits_linux.cpp:75:38: error: static assertion failed
[ALL  ]       75 | COMPILER_CHECK(struct_kernel_stat_sz == sizeof(struct stat));
[ALL  ]          |                ~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
[ALL  ]    /home/bagas/cross/workdir/mips64-unknown/.build/mips64-unknown-linux-gnu/src/gcc/libsanitizer/sanitizer_common/sanitizer_internal_defs.h:332:44: note: in definition of macro 'COMPILER_CHECK'
[ALL  ]      332 | #define COMPILER_CHECK(pred) static_assert(pred, "")
[ALL  ]          |                                            ^~~~
[ALL  ]    mv -f .deps/sanitizer_platform_limits_openbsd.Tpo .deps/sanitizer_platform_limits_openbsd.Plo
[ERROR]    make[5]: *** [Makefile:616: sanitizer_platform_limits_linux.lo] Error 1
```

--- a/packages/gcc/11.3.0/0010-libsanitizer-mips64-Set-struct_kernel_stat_sz-to-mat.patch
+++ b/packages/gcc/11.3.0/0010-libsanitizer-mips64-Set-struct_kernel_stat_sz-to-mat.patch
@@ -1,0 +1,31 @@
+From ac6e15a2776e8fffdbd108213e8690c9f30a7380 Mon Sep 17 00:00:00 2001
+From: Chris Packham <chris.packham@alliedtelesis.co.nz>
+Date: Mon, 16 May 2022 20:55:36 +1200
+Subject: [PATCH] libsanitizer: mips64: Set struct_kernel_stat_sz to match the
+ kernel
+
+The kernel's struct stat is 104 bytes when compiling for
+_MIPS_SIM_ABI64. Set struct_kernel_stat_sz to match.
+
+Signed-off-by: Chris Packham <chris.packham@alliedtelesis.co.nz>
+Link: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=105614
+---
+ libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.h | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.h b/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.h
+index 83861105a509..326419a5d613 100644
+--- a/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.h
++++ b/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.h
+@@ -83,7 +83,7 @@ const unsigned struct_kernel_stat64_sz = 104;
+ #elif defined(__mips__)
+ const unsigned struct_kernel_stat_sz = SANITIZER_ANDROID
+                                            ? FIRST_32_SECOND_64(104, 128)
+-                                           : FIRST_32_SECOND_64(144, 216);
++                                           : FIRST_32_SECOND_64(144, 104);
+ const unsigned struct_kernel_stat64_sz = 104;
+ #elif defined(__s390__) && !defined(__s390x__)
+ const unsigned struct_kernel_stat_sz = 64;
+-- 
+2.36.1
+

